src/OrchardCore/OrchardCore.DisplayManagement.Liquid/OrchardCoreBuilderExtensions.cs

```diff
@@ -71,25 +71,7 @@ public static OrchardCoreBuilder AddLiquidViews(this OrchardCoreBuilder builder)
 
                 o.Scope.SetValue("Culture", new CultureValue());
 
-                o.Scope.SetValue("Environment", new ObjectValue(new LiquidEnvironmentAccessor()));
-                o.MemberAccessStrategy.Register<LiquidEnvironmentAccessor, FluidValue>((obj, name, ctx) =>
-                {
-                    var hostEnvironment = ((LiquidTemplateContext)ctx).Services.GetRequiredService<IHostEnvironment>();
-
-                    if (hostEnvironment != null)
-                    {
-                        return name switch
-                        {
-                            "IsDevelopment" => BooleanValue.Create(hostEnvironment.IsDevelopment()),
-                            "IsStaging" => BooleanValue.Create(hostEnvironment.IsStaging()),
-                            "IsProduction" => BooleanValue.Create(hostEnvironment.IsProduction()),
-                            "Name" => StringValue.Create(hostEnvironment.EnvironmentName),
-                            _ => NilValue.Instance
-                        };
-                    }
-
-                    return NilValue.Instance;
-                });
+                o.Scope.SetValue("Environment", new HostingEnvironmentValue());
```

Comment:
I go with the current implement because I can't resolve `IHostEnvironment`, let me check again. Agree that resolving the value once is enough in case of the environment not like the culture